src/impl_methods.rs

```diff
@@ -214,11 +218,14 @@ impl<A, S, D> ArrayBase<S, D> where S: Data<Elem=A>, D: Dimension
     /// [`D::SliceArg`]: trait.Dimension.html#associatedtype.SliceArg
     ///
     /// **Panics** if an index is out of bounds or stride is zero.<br>
-    /// (**Panics** if `D` is `IxDyn` and `indexes` does not match the number of array axes.)
-    pub fn slice(&self, indexes: &D::SliceArg) -> ArrayView<A, D> {
-        let mut arr = self.view();
-        arr.islice(indexes);
-        arr
+    /// (**Panics** if `D` is `IxDyn` and `info` does not match the number of array axes.)
+    pub fn slice<I, T, Do>(&self, info: I) -> ArrayView<A, Do>
+    where
+        I: Borrow<SliceInfo<T, Do>>,
+        T: Borrow<D::SliceArg>,
```

Comment:
- I would like to avoid the T parameter and simplify the types here, ideally using `SliceInfo<D::SliceArg, ...`, but I see that it's not exactly easy.

Reply:
Hmm... Removing the `T` parameter is difficult because `s!` always returns a `SliceInfo<T, D>` where `T` is a fixed-length array. The constraint could be `I: Borrow<SliceInfo<D::SliceArg, Do>>` for `Ix0` to `Ix6`, but this wouldn't work for `IxDyn` because `IxDyn::SliceArg = [SliceOrIndex]`. There are a few possible alternatives:
* `slice()` could be implemented separately for each dimension `D`, so for `Ix0` to `Ix6`, the `T` parameter wouldn't be necessary.
* I think it would be possible to change the constraint to `&'a I: Into<SliceInfo<&'b D::SliceArg, Do>>`, but that adds lifetime parameters, which IMHO are worse than a type parameter.
* If `IxDyn::SliceArg = Vec<SliceOrIndex>`, then I think it would be possible to change the constraint to `I: Into<SliceInfo<D::SliceArg, Do>>`, but that would mean that `slice()` would take ownership of the `info` parameter, which is not desirable.
* If `IxDyn::SliceArg = Vec<SliceOrIndex>`, then I think it would be possible to change the constraint to `&'a I: Into<SliceInfo<D::SliceArg, Do>>`, but that would require the `info` argument to be a reference, which is not really desirable (see below).

It would be possible to remove the `I` parameter and instead make the signature `pub fn slice<T, Do>(&self, info: &SliceInfo<T, Do>) -> ArrayView<A, Do>`. However, that would mean that `info` has to be a reference, which would make usage with the `s!` macro less clean: `arr.slice(&s![1..4, 3])`. (It would be possible to make the `s!` macro to return `&SliceInfo` instead of `SliceInfo`, but as far as I know, then the `&SliceInfo` couldn't leave the scope where it was created, so it couldn't be returned from a function.)

In conclusion, in my limited Rust knowledge, the best options are requiring `slice(&s![])` or requiring the extra `I` parameter. If you'd like me to remove `I` and require `slice(&s![])`, or if you have any other ideas, please let me know.

Reply:
I will look at this in detail late. Looks possible to use `&SliceInfo< [SliceorIndex]>` for the ixdyn case, as long as the macro makes that but a fixed size array instead.